### PR TITLE
wd/sec: add new algorithms for Kunpeng930

### PR DESCRIPTION
**Pull request by Liulongfang**
Add new BD type3 algorithms for SEC driver
and solved the driver compatibility issues.

Signed-off-by: Liulongfang <836713992@qq.com>

### CODE REVIEW

**Comment by hzhuang1** [comment]
The v3 is just copied from v2. So there's no difference between v2 and v3. Why do we need this patch?

**Comment by hzhuang1** [comment]
Sorry. I just clicked the wrong button. So re-open it.

**Comment by Liulongfang** [comment]
> The v3 is just copied from v2. So there's no difference between v2 and v3. Why do we need this patch?

no, the driver of v3 is completely different from v2.

File: drv/hisi_qm_udrv.c [hunk 1]
```diff
@@ -75,6 +75,10 @@ static struct hisi_qm_type qm_type[] = {
 		.qm_name	= "hisi_qm_v2",
 		.qm_db_offs	= QM_V2_DOORBELL_OFFSET,
 		.hacc_db	= hacc_db_v2,
+	}, {
+		.qm_name	= "hisi_qm_v3",
+		.qm_db_offs	= QM_V2_DOORBELL_OFFSET,
+		.hacc_db	= hacc_db_v2,
```
Comment (wangzhou) [inline drv/hisi_qm_udrv.c:81]: I think this is wrong, I should think hardware will not change in futurn,  so v2 can run for futurn v3, v4, v5.

Comment (Liulongfang) [reply drv/hisi_qm_udrv.c:81]: I need the version handle to compatible interface：

**Pull request closed without merging**